### PR TITLE
fix(editor): end the link when a space is typed at its end

### PR DESCRIPTION
**Pull request by netchampfaris**
Typing a space right after a link pulled the space inside the `<a>`, and everything typed after it stayed linked — so a link grew to swallow the rest of the sentence one keystroke at a time:

```
before:  <p>see <a href="…">docs</a></p>   + " and more"
         → <p>see <a href="…">docs and more</a></p>
after:   → <p>see <a href="…">docs</a> and more</p>
```

## Why

`@tiptap/extension-link` declares `inclusive() { return this.options.autolink }`, and we enable `autolink`. An inclusive mark extends over text typed at its end — that's deliberate, so the mark can keep up with a URL that's still being typed. It also means the space that ends the URL gets marked, and every character after it inherits the mark from the linked space before it.

The existing `clear-link-on-boundary-plugin` can't catch this: it only clears `storedMarks`, and here the mark comes from the document itself.

## Fix

New `exit-link-on-space-plugin.ts`, registered alongside the other link plugins. Whitespace is where a URL unambiguously ends, so that's the boundary it acts on: after a doc change, if the collapsed cursor sits at the end of a link preceded by a run of linked whitespace, the link mark is removed from that run. Text typed next inherits the now-unlinked space's marks, which ends the link — without weakening the inclusive-mark behaviour autolink relies on mid-URL.

Deliberate non-changes:
- **A space typed *inside* link text stays linked** (`click|me` → `click me` remains one link). Splitting a link in two is never what that keystroke meant, so the plugin bails when the same link mark continues past the cursor.
- **Non-whitespace typed at the end of a link still extends it** (`docs` + `/api` → `docs/api`), which is what makes autolink usable while a URL is mid-typing.

It runs as `appendTransaction` rather than `handleTextInput` so input rules keep first crack at the keystroke, and so pasted or IME-composed whitespace is covered too.

`experimental/TextEditor` shares the same defaults but is marked frozen v0 code, so it's untouched.

## Edge cases

Probed against a real editor; the first two turned up regressions in the first pass of this PR, now fixed and covered by tests:

| Case | Behaviour |
| --- | --- |
| Link applied to a whitespace-only selection | Link kept. Ending a link is the plugin's job, deleting one is not — it bails when the whitespace run starts at the mark's own start. |
| Doc change *elsewhere* (collaborator keystroke, undo) with the cursor resting after an already-linked space | Marks untouched. The whitespace must overlap a changed range, computed with the same `combineTransactionSteps`/`getChangedRanges` helpers tiptap's own autolink uses. |
| Undo after typing the space | One step — history groups the appended transaction with the keystroke. |
| Loaded content with a linked trailing space (`setContent`) | Never rewritten; no spurious update event. |
| Selecting a link and typing over it | Unchanged. |
| A second URL typed after an existing link | Still autolinked. |
| `getHTML()` round-trip | Idempotent. |

Because the changed-range check reads mark-only steps too, applying a link over a selection that includes a trailing space now leaves that space outside the `<a>` — which is the underlined trailing space visible in the report.

## Why tiptap ships it this way

It's a tradeoff, not an oversight. `inclusive` is what lets the mark keep up with link text that's still being edited: without it, appending to an existing link leaves the new characters outside it (`<a>example.com</a>/docs`). Tiptap ties it to `autolink` and offers `exitable: true` as the escape hatch — which binds **ArrowRight** to `Mark.handleExit`, removing the stored mark and inserting a space for you. That only fires at the end of a textblock, and nobody discovers it. Typing a space, which is what users actually do, was left to fall through to inclusive-mark behaviour.

## Testing

`exit-link-on-space-plugin.test.ts` — 12 cases against a real `Editor`, typing character by character through `view.dispatch`: the space at a link end, following text staying unlinked, runs of spaces, a link followed by existing text, the mid-link space that must stay linked, non-whitespace still extending, autolink not relinking the space that triggered it, and the read-only guard.

- `vitest run src/molecules/editor` — 27 files, 193 tests pass
- `vue-tsc --noEmit` — no new errors

🤖 Generated with [Claude Code](https://claude.com/claude-code)

Docs preview: https://ui.frappe.io/pr-preview/pr-1060/

Coverage: 71.79% (+0.08% vs `main`)

### CODE REVIEW

**Comment by netchampfaris** [comment]
Both addressed in 230250160.

**Greptile (P1, remote updates):** valid, and the sharper version of what barista raised. The changed-range guard stops an edit *elsewhere* from rewriting marks at the cursor, but a peer typing the space at that same spot still triggered a local exit — pushing our normalisation into their typing, on top of the exit their own editor already replicated.

`list-join` had this guard for the same reason, so rather than duplicating twenty lines of y-prosemirror plugin-key sniffing I extracted `isRemoteChange` into `shared/collaboration.ts` and used it in both. New test drives it through a `y-sync` plugin stand-in, matching the existing `list-join` test. Moving the helper also types `PluginKey`'s runtime `key` field explicitly, which drops a pre-existing type-check error in `list-join`.

**barista (nit 1, gate is "some transaction changed the doc"):** you reviewed 6920f5dbd; 8336e398f had already narrowed it to "the changed range covers `range`", via the same `combineTransactionSteps`/`getChangedRanges` helpers tiptap's own autolink uses. Two things fell out of that:

- Your undo scenario now behaves: with `<a>docs </a>tail`, typing `y` at the link end and undoing lands back on the original, because the `removeMark` no longer fires on a change that didn't touch the whitespace. There's a test for undo being a single step.
- Because the check reads mark-only steps too, applying a link over a selection with a trailing space leaves that space outside the `<a>` — the underlined trailing space from the report.

That pass also caught something neither of you saw: applying a link to a whitespace-only selection made the link vanish outright, since "strip the trailing whitespace" was the whole mark. It now bails when the run starts at the mark's own start — ending a link is this plugin's job, deleting one isn't.

**barista (nit 2, read-only test path):** agreed, and left as-is deliberately — a read-only user can't type, so `view.dispatch` is the only way in. Added a comment saying so, since what it actually pins is a programmatic write into a read-only editor.

13 link tests; full suite 1165 pass.

File: src/molecules/editor/extensions/link/exit-link-on-space-plugin.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  options: ExitLinkOnSpacePluginOptions,
+): Plugin {
+  return new Plugin({
+    key: new PluginKey('exitLinkOnSpace'),
+    appendTransaction: (transactions, _oldState, newState) => {
+      if (!options.editor.isEditable) {
+        return null
+      }
+
+      if (!transactions.some((transaction) => transaction.docChanged)) {
```
[the remark this answers is not in the document]
Comment (netchampfaris) [reply src/molecules/editor/extensions/link/exit-link-on-space-plugin.ts:48]: Valid — fixed in 230250160. Extracted `list-join`'s `isRemoteChange` into `shared/collaboration.ts` (rather than duplicating the y-prosemirror plugin-key sniffing) and gated this plugin on it too. A remote peer's space is that peer's link to end: their own editor runs this same plugin and the result replicates, so acting here as well pushed a local rewrite into their typing. Covered by a test using a `y-sync` plugin stand-in, matching the existing `list-join` test.
